managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -355,9 +355,17 @@ public void operationComplete(ManagedLedgerInfo mlInfo, Stat stat) {
                                         .setTimestamp(clock.millis()).build();
                                 ledgers.put(id, info);
                                 if (managedLedgerInterceptor != null) {
-                                    managedLedgerInterceptor.onManagedLedgerLastLedgerInitialize(name, lh);
+                                    managedLedgerInterceptor.onManagedLedgerLastLedgerInitialize(name, lh)
+                                        .whenComplete((__, ex) -> {
+                                            if (ex != null) {
+                                                callback.initializeFailed(new ManagedLedgerInterceptException(ex));
+                                            } else {
+                                                initializeBookKeeper(callback);
+                                            }
+                                        });
```

Comment:
In general it's preferable to do: 
```suggestion
                                        .thenRun(() -> initializeBookKeeper(callback))
                                        .exceptionally(ex -> {
                                            callback.initializeFailed(new ManagedLedgerInterceptException(ex));
                                            return null;
                                        });
```

The reason is that if there is any unexpected exception thrown when calling `initializeBookKeeper()`, the exceptionally part will be invoked.